go/oasis-test-runner/scenario/e2e/runtime/history_reindex.go

```diff
@@ -120,3 +159,77 @@ func (sc *historyReindexImpl) registerRuntime(ctx context.Context, childEnv *env
 	rtDsc.Deployments[0].ValidFrom = epoch + 1
 	return sc.RegisterRuntime(childEnv, cli, rtDsc, 0)
 }
+
+func (sc *historyReindexImpl) fetchRuntimeBlock(ctx context.Context, client api.RuntimeClient, round uint64) (*block.Block, error) {
+	blk, err := client.GetBlock(ctx, &api.GetBlockRequest{
+		RuntimeID: KeyValueRuntimeID,
+		Round:     round,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get block: %w", err)
+	}
+	return blk, nil
+}
+
+func (sc *historyReindexImpl) verifyReindexedBlocks(ctx context.Context, client *oasis.Client, endRound uint64) error {
+	compute := sc.Net.ComputeWorkers()[0]
+	ctrl1, err := oasis.NewController(compute.SocketPath())
+	if err != nil {
+		return fmt.Errorf("failed to create controller (node: %s): %w", compute.Name, err)
+	}
+	ctrl2, err := oasis.NewController(client.SocketPath())
+	if err != nil {
+		return fmt.Errorf("failed to create controller (node: %s): %w", client.Name, err)
+	}
+	indexClient, reindexClient := ctrl1.RuntimeClient, ctrl2.RuntimeClient
```

Comment:
These names are confusing as before you used suffixes 1/2, and now you are using prefixes index/reindex. Options: `client1`, `client2`, or `computeCtrl`, `clientCtrl`, `comptuteRt`, `clientRt` , ...